Source/Core/DolphinWX/LuaScriptFrame.cpp

```diff
@@ -0,0 +1,553 @@
+int setCStick(lua_State* L);
+int setCStickPolar(lua_State* L);
+int getButtons(lua_State* L);
+int setButtons(lua_State* L);
+int setDPad(lua_State* L);
+int getTriggers(lua_State* L);
+int setTriggers(lua_State* L);
+
+// GLOBAL IS NECESSARY FOR LOG TO WORK
+LuaScriptFrame* currentWindow;
+
+// External symbols
+std::map<const char*, LuaFunction>* registered_functions;
+std::mutex lua_mutex;
+
+LuaScriptFrame::LuaScriptFrame(wxWindow* parent)
+    : wxFrame(parent, wxID_ANY, _("Lua Console"), wxDefaultPosition, wxSize(431, 397),
+              wxDEFAULT_FRAME_STYLE ^ wxRESIZE_BORDER)
+{
+  CreateGUI();
+  SetIcons(WxUtils::GetDolphinIconBundle());
+
+  Center();
+  Show();
+
+  currentWindow = this;
+  lua_thread = nullptr;
+
+  // Create function map if it doesn't already exist
+  if (!registered_functions)
+  {
+    registered_functions = new std::map<const char*, LuaFunction>;
+
+    registered_functions->insert(std::pair<const char*, LuaFunction>("print", printToTextCtrl));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("frameAdvance", frameAdvance));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("getFrameCount", getFrameCount));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("getMovieLength", getMovieLength));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("softReset", softReset));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("saveState", saveState));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("loadState", loadState));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("setEmulatorSpeed", setEmulatorSpeed));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getAnalog", getAnalog));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setAnalog", setAnalog));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getCStick", getCStick));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setCStick", setCStick));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("setCStickPolar", setCStickPolar));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("setAnalogPolar", setAnalogPolar));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getButtons", getButtons));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setButtons", setButtons));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setDPad", setDPad));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getTriggers", getTriggers));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setTriggers", setTriggers));
+  }
+}
+
+LuaScriptFrame::~LuaScriptFrame()
+{
+  // Disconnect Events
+  this->Disconnect(wxID_ANY, wxEVT_COMMAND_MENU_SELECTED,

```

Comment:
These explicit `Disconnects` are likely not necessary, as the internal event table is automatically destroyed when the frame is destroyed.